### PR TITLE
New iteration on scrollbars

### PR DESCRIPTION
**Pull request by hamen**
# Overview

This PR is the first iteration of the new scrollbar style. It serves as a cut down version of #457. It contains:

* updated `IntUiBridge` and the new `ScrollbarBridge` for in-IDE development
* updated `IntUiScrollbarStyling`
* new Scrollbars showcase screen in Standalone (see GIF below)

Recording2024-08-05123245-ezgif com-video-to-gif-converter

# Further development

* Tab scrollbars need more love #511
* We need a mechanism to update scrollbar style on the fly on macOS #510  
* TextAreas will have scrollbars in the upcoming PR that covers #108

### CODE REVIEW

File: ide-laf-bridge/src/main/kotlin/org/jetbrains/jewel/bridge/theme/IntUiBridge.kt [hunk 1]
```diff
@@ -706,7 +704,7 @@ private fun readMenuStyle(): MenuStyle {
                 contentPadding = PaddingValues(horizontal = 0.dp, vertical = 6.dp),
                 offset = DpOffset(0.dp, 2.dp),
                 shadowSize = 12.dp,
-                borderWidth = retrieveIntAsDpOrUnspecified("Popup.borderWidth").takeOrElse { 1.dp },
+                borderWidth = retrieveIntAsDpOrUnspecified("Popup.borderWidth").takeOrElse { 2.dp },
```
Comment (rock3r) [inline ide-laf-bridge/src/main/kotlin/org/jetbrains/jewel/bridge/theme/IntUiBridge.kt:707]: This is a regression on #452 ❌

File: ide-laf-bridge/src/main/kotlin/org/jetbrains/jewel/bridge/theme/ScrollbarBridge.kt [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+            thumbThicknessExpanded = 8.dp,
+            minThumbLength = 16.dp,
+            trackPadding = PaddingValues(),
+            trackPaddingExpanded = PaddingValues(),
+        )
+    }
+
+private fun readMacScrollbarStyle(): ScrollbarVisibility {
+    require(hostOs.isMacOS) { "You shouldn't be calling this on Win/Linux" }
+
```
Comment (rock3r) [inline ide-laf-bridge/src/main/kotlin/org/jetbrains/jewel/bridge/theme/ScrollbarBridge.kt:199]: Nobody calls this directly, as it's private, so... I'd remove these require calls. It'll blow up anyway if this happens by mistake on Windows/Linux.

```suggestion
```

File: ide-laf-bridge/src/main/kotlin/org/jetbrains/jewel/bridge/theme/ScrollbarBridge.kt [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+            ScrollbarVisibility.WhenScrolling.Companion.defaults()
+        } else {
+            ScrollbarVisibility.AlwaysVisible
+        }
+    return visibility
+}
+
+private fun readMacScrollbarBehavior(): TrackClickBehavior {
+    require(hostOs.isMacOS) { "You shouldn't be calling this on Win/Linux" }
+
```
Comment (rock3r) [inline ide-laf-bridge/src/main/kotlin/org/jetbrains/jewel/bridge/theme/ScrollbarBridge.kt:215]:
```suggestion
```

File: ui/src/main/kotlin/org/jetbrains/jewel/ui/component/Scrollbars.kt [hunk 4]
```diff
@@ -60,60 +251,461 @@ public fun HorizontalScrollbar(
...
+                    style.colors.thumbBackgroundHovered
+                },
+                animationSpec = tween(durationMillis = style.scrollbarVisibility.expandAnimationDuration.inWholeMilliseconds.toInt()),
+            ).value
+        } else {
+            if (isHighlighted) {
+                style.colors.thumbBackgroundHovered
+            } else {
+                style.colors.thumbBackgroundHovered
+            }
```
Comment (rock3r) [inline ui/src/main/kotlin/org/jetbrains/jewel/ui/component/Scrollbars.kt:356]: The inner if-else look suspicious, they return the hovered colour in both cases

File: ui/src/main/kotlin/org/jetbrains/jewel/ui/component/TabStrip.kt [hunk 5]
```diff
@@ -42,24 +44,25 @@ public fun TabStrip(
...
+            Modifier
+                .horizontalScroll(scrollState)
+                .scrollable(
+                    orientation = Orientation.Vertical,
+                    reverseDirection =
+                    ScrollableDefaults.reverseDirection(
+                        LocalLayoutDirection.current,
+                        Orientation.Vertical,
+                        false,
+                    ),
```
Comment (rock3r) [inline ui/src/main/kotlin/org/jetbrains/jewel/ui/component/TabStrip.kt:62]: Indent seems wrong here

**Review by rock3r — CHANGES_REQUESTED** [review]

File: ide-laf-bridge/src/main/kotlin/org/jetbrains/jewel/bridge/theme/IntUiBridge.kt (hunk 1 above)
Comment (hamen) [reply ide-laf-bridge/src/main/kotlin/org/jetbrains/jewel/bridge/theme/IntUiBridge.kt:707]: Fixed in bb9b0de0

File: ide-laf-bridge/src/main/kotlin/org/jetbrains/jewel/bridge/theme/ScrollbarBridge.kt (hunk 3 above)
Comment (hamen) [reply ide-laf-bridge/src/main/kotlin/org/jetbrains/jewel/bridge/theme/ScrollbarBridge.kt:215]: Done in 5d33a57f40f1ab9b66a994b999f394b739c4d927

File: ui/src/main/kotlin/org/jetbrains/jewel/ui/component/Scrollbars.kt (hunk 4 above)
Comment (hamen) [reply ui/src/main/kotlin/org/jetbrains/jewel/ui/component/Scrollbars.kt:356]: Nice catch. Fixed in cec665f166f761f269c21247f741bd0ded48713b

**Review by rock3r — APPROVED** [review]

**Pull request merged**